File: src/main/java/bbarena/model/event/game/BallEvent.java
package bbarena.model.event.game;

import bbarena.model.Arena;
import bbarena.model.event.Event;
import bbarena.model.pitch.Ball;

public abstract class BallEvent extends Event {

	private static final long serialVersionUID = -5393891334745678203L;
	protected int _ballId = 0;

	public BallEvent(int ballId) {
		super();
		_ballId = ballId;
	}

	public int getBallId() {
		return _ballId;
	}

	public Ball getBall(Arena arena) {
		return arena.getPitch().getBall(_ballId);
	}

}

File: src/main/kotlin/bbarena/model/event/game/BallEvent.kt
package bbarena.model.event.game

import bbarena.model.Arena
import bbarena.model.event.Event
import bbarena.model.pitch.Ball

abstract class BallEvent(val ballId: Int = 0) : Event() {

    fun getBall(arena: Arena): Ball {
        return arena.pitch.getBall(ballId)
    }

}